### PR TITLE
Use cloudscraper for Teamskeet

### PR DESCRIPTION
**Pull request by bnkai**
From discord it seems that the api endpoint used by teamskeet uses now cloudflare
I replaced requests with cloudscraper, it will need some testing though

### CODE REVIEW

File: scrapers/TeamskeetAPI.py [hunk 1]
```diff
@@ -39,63 +33,77 @@ def save_json(api_json, url):
...
+    try:
+        scene_api_json_check = r.json().get('found')
+        if scene_api_json_check:
+            scene_api_json = r.json()['_source']
+        else:
+            log.error('Scene not found (Wrong ID?)')
+            sys.exit(1)
+
+    except:
+        log.debug(f"invalid input {r.content}")
```
Comment (Belleyy) [inline scrapers/TeamskeetAPI.py:97]:
```suggestion
        if "Please Wait... | Cloudflare" in r.text:
            log.error("Protected by Cloudflare.")
        else:
            log.debug(f"invalid input {r.content}")
```
I am not a fan of printing the whole html into the log.

File: scrapers/TeamskeetAPI.py [hunk 2]
```diff
@@ -39,63 +33,77 @@ def save_json(api_json, url):
...
         'Referer': 'https://www.teamskeet.com/'
     }
-
+    scraper = cloudscraper.create_scraper()
     # Send to the API
     r = ""
     try:
-        r = requests.get(api_url, headers=headers, timeout=(3, 5))
+        r = scraper.get(api_url, headers=headers, timeout=(3, 5))
     except requests.exceptions.RequestException:
```
Comment (Belleyy) [inline scrapers/TeamskeetAPI.py:80]:
```suggestion
    except cloudscraper.exceptions.CloudflareException:
```
It's not request for sure, but not sure if this one is the correct exception.

File: scrapers/TeamskeetAPI.py [hunk 3]
```diff
@@ -39,63 +33,77 @@ def save_json(api_json, url):
...
-        print("An error has occurred with Requests", file=sys.stderr)
-        print(f"Request status: `{r.status_code}`", file=sys.stderr)
-        print(f"Check your TeamskeetJSON.log for more details", file=sys.stderr)
+        log.error("An error has occurred with Requests")
+        log.error(f"Request status: `{r.status_code}`")
+        log.error(f"Check your TeamskeetJSON.log for more details")
```
Comment (Belleyy) [inline scrapers/TeamskeetAPI.py:83]:
```suggestion
        log.error("Check your TeamskeetJSON.log for more details")
```

File: scrapers/TeamskeetAPI.py (hunk 1 above)
Comment (bnkai) [reply scrapers/TeamskeetAPI.py:97]: yikes..yes i only used that to see why the scraper was blocking, i was going to remove that later on.
should we skip the `r.content` print in the else also and leave a simple `log.error("invalid input")` ?

Comment (Belleyy) [reply scrapers/TeamskeetAPI.py:97]: I would keep the `log.error("invalid input")`

File: scrapers/TeamskeetAPI.py (hunk 2 above)
Comment (Belleyy) [reply scrapers/TeamskeetAPI.py:80]: yea sure

**Comment by bnkai** [comment]
Updated with latest suggestions
I tried to get CF to block me uising the scraper. It took 5 scrapes within one minute to do that and a couple of minutes (without retyring inbetween) to get unblocked.

**Pull request merged**